classes/robot/crawler.php

```diff
@@ -143,6 +143,8 @@ public function absolute_url($base, $rel) {
         $host = $parts['host'];
 
         // Remove non-directory element from path.
+        // This is converting anything like '/XYZ' to '' except of '//' and '///' and '//////' and etc.
+        // It seems to be wrong.
```

Comment:
Either way you should also add some tests using a bases like:

'base' => 'http://test.com/sub1/sub2/file1.php',

which is the proper way to exercise this bit of code